### PR TITLE
Malf no longer ends the round when the AI is destroyed

### PR DESCRIPTION
**Pull request by MadmanMartian**
You've destroyed the rogue AI hellbent on destroying the station. Get back to work.

### CODE REVIEW

**Comment by DrSnips** [comment]
While it fits datums fine I do worry about how a lot of the crew is likely dead or borged/an mmi if blown. Malf either ends with JUSTing the station (station bacon, infinite silicon works, etc) or fucking nothing happening and I'd really hate to be stuck on a station with the former.

File: code/datums/gamemode/factions/malf.dm [hunk 1]
```diff
@@ -27,7 +27,33 @@
...
+				if (R.antag.current && !R.antag.current.isDead())
+					is_ded = FALSE
+					break
+		if(is_ded)
+			malf_mode_declared = FALSE
+			set_security_level(SEC_LEVEL_GREEN)
+			world << sound('sound/misc/notice1.ogg')
+			command_alert(/datum/command_alert/malf_destroyed)
+			var/interceptname = "Malfunctioning AI lockdown lifted"
+			var/intercepttext = {"<Font size = 3><B>Nanotrasen Update</B>: Biohazard contained.</FONT><HR>
```
Comment (Kurfursten) [inline code/datums/gamemode/factions/malf.dm]: biohazard?

**Comment by Ephx1** [comment]
Make destroying the Malf AI call the shuttle. No problem, recall. Station fucked, let it ride.

**Comment by Kurfursten** [comment]
This is a less OOP and malfcentric version of #21999

**Comment by ShiftyRail** [comment]
Then #21999 should be merged.
The non-technical aspects of it (ie, whether or not malf should call a shuttle like Cult does) can be discussed later on, as those would be easier to implement.

File: code/datums/gamemode/factions/malf.dm [hunk 2]
```diff
@@ -27,7 +27,33 @@
 /datum/faction/malf/process()
 	if(apcs >= 3 && malf_mode_declared && can_malf_ai_takeover())
 		AI_win_timeleft -= ((apcs / 6) * SSticker.getLastTickerTimeDuration()) //Victory timer de-increments based on how many APCs are hacked.
-
+	if(malf_mode_declared)
+		var/is_ded = TRUE
```
Comment (ShiftyRail) [inline code/datums/gamemode/factions/malf.dm:31]: I've been told to change a define that was just "MAGBOOTIES" so I'll nitpick.

**Comment by SonixApache** [comment]
\*revives malf*
\*gamemode breaks*

File: code/datums/gamemode/factions/malf.dm (hunk 1 above)
Comment (MadmanMartian) [reply code/datums/gamemode/factions/malf.dm]: whoops

**Comment by MadmanMartian** [comment]
going to refactor to use #21999 's new faction stages

**Comment by Kurfursten** [comment]
Already implements this functionality. I also stole your malf destroyed message

**Comment by MadmanMartian** [comment]
In fact, it's outright handled by 21999, so RIP this PR

**Pull request closed without merging**